### PR TITLE
Remote data access tutorial for CMIP6 Zarr data

### LINKED ISSUE #103: Add intermediate remote data access tutorial
Illustrate
- [ ] opendap / thredds
- [ ] datasets on S3
- [ ] datasets using gcsfs
- [ ] fsspec

Example code:
``` python
import gcsfs

fs = gcsfs.GCSFileSystem(token="anon")
ds = xr.open_zarr(
    fs.get_mapper("gs://pangeo-noaa-ncei/noaa.ersst.v5.zarr"), consolidated=True
).load()
ds
```

### PR DESCRIPTION
**Pull request by weiji14**
Draft tutorial on accessing cloud-native CMIP6 datasets on Google Cloud. Covers opening Zarr files using xarray.open_zarr. Also did a basic sea level change (between 2015 and 2100) calculation and plot to show the result.

Tutorial adapted from https://colab.research.google.com/drive/19iEVxE_9QoTeg4st7MmucHJUmO93NXHp?usp=sharing#scrollTo=4M8ypd9jOzh9

Might pay if a sea level rise expert double checks if the calculation is ok. I feel like the results still differ from what's shown at the [IPCC interactive map](https://interactive-atlas.ipcc.ch/regional-information#eyJ0eXBlIjoiQVRMQVMiLCJjb21tb25zIjp7ImxhdCI6ODQwNzEwOSwibG5nIjotNTMwOTg3OCwiem9vbSI6MywicHJvaiI6IkVQU0c6NTQwMzAiLCJtb2RlIjoiY29tcGxldGVfYXRsYXMifSwicHJpbWFyeSI6eyJzY2VuYXJpbyI6InNzcDU4NSIsInBlcmlvZCI6ImZhciIsInNlYXNvbiI6InllYXIiLCJkYXRhc2V0IjoiQ01JUDYiLCJ2YXJpYWJsZSI6InNsciIsInZhbHVlVHlwZSI6IkFOT01BTFkiLCJoYXRjaGluZyI6IkNPTVBSRUhFTlNJVkUiLCJyZWdpb25TZXQiOiJhcjYiLCJiYXNlbGluZSI6IkFSNiIsInJlZ2lvbnNTZWxlY3RlZCI6W119LCJwbG90Ijp7ImFjdGl2ZVRhYiI6InBsdW1lIiwibWFzayI6Im5vbmUiLCJzY2F0dGVyWU1hZyI6bnVsbCwic2NhdHRlcllWYXIiOm51bGwsInNob3dpbmciOmZhbHNlfX0=)

Fixes #103.

### CODE REVIEW

**Comment by dcherian** [comment]
@jbusecke would you mind doing a review here?

File: intermediate/remote_data_access.ipynb [hunk 1]
```diff
@@ -0,0 +1,331 @@
+{
```
Comment (jbusecke) [inline intermediate/remote_data_access.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    df = pd.read_csv("https://storage.googleapis.com/cmip6/cmip6-zarr-consolidated-stores.csv")
</pre><p>This is potentially an outdated catalog file. I need to "clean house" at some point, but in the meantime this is the "officially" maintained csv: "https://cmip6.storage.googleapis.com/pangeo-cmip6.csv"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xarray-contrib/xarray-tutorial/pull/132/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (jbusecke) [inline intermediate/remote_data_access.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    zos_2015jan = ds.zos.sel(time="2015-01-16").isel(time=0)
</pre><p>Why the double selection? Shouldn't the first <code>.sel()</code> do the trick? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xarray-contrib/xarray-tutorial/pull/132/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Comment by jbusecke** [comment]
Nice example @weiji14

> Might pay if a sea level rise expert double checks if the calculation is ok. I feel like the results still differ from what's shown at the [IPCC interactive map](https://interactive-atlas.ipcc.ch/regional-information#eyJ0eXBlIjoiQVRMQVMiLCJjb21tb25zIjp7ImxhdCI6ODQwNzEwOSwibG5nIjotNTMwOTg3OCwiem9vbSI6MywicHJvaiI6IkVQU0c6NTQwMzAiLCJtb2RlIjoiY29tcGxldGVfYXRsYXMifSwicHJpbWFyeSI6eyJzY2VuYXJpbyI6InNzcDU4NSIsInBlcmlvZCI6ImZhciIsInNlYXNvbiI6InllYXIiLCJkYXRhc2V0IjoiQ01JUDYiLCJ2YXJpYWJsZSI6InNsciIsInZhbHVlVHlwZSI6IkFOT01BTFkiLCJoYXRjaGluZyI6IkNPTVBSRUhFTlNJVkUiLCJyZWdpb25TZXQiOiJhcjYiLCJiYXNlbGluZSI6IkFSNiIsInJlZ2lvbnNTZWxlY3RlZCI6W119LCJwbG90Ijp7ImFjdGl2ZVRhYiI6InBsdW1lIiwibWFzayI6Im5vbmUiLCJzY2F0dGVyWU1hZyI6bnVsbCwic2NhdHRlcllWYXIiOm51bGwsInNob3dpbmciOmZhbHNlfX0=)

Not a sea level expert at all, but the fact that this is only one experiment might explain the discrepancy. I would be happy to extend this example for more models/members (similar to what I have done for [Scipy](https://github.com/jbusecke/presentation_scipy_2022_cmip)).

Also reading in the data is shown nicely here, but it is only one of many ways (this really needs some improvement from our side). A suggestion would be to point to the [official pangeo docs for the data](https://pangeo-data.github.io/pangeo-cmip6-cloud/accessing_data.html#manually-searching-the-catalog), so that if we update this in the future this example points to the updated methodology?

File: intermediate/remote_data_access.ipynb (hunk 1 above)
Comment (dcherian) [reply intermediate/remote_data_access.ipynb:1]: Yeah this is a weirdness around `.sel` with datetimes. I would stick `.squeeze()` at the end maybe. Or replace with `.sel(time="2015-01-16 00:00")`

**Comment by dcherian** [comment]
Thanks for the review @jbusecke 

> I would be happy to extend this example for more models/member

I think not necessary here; we could just link to your efforts at easy_ipcc (cool repo name btw). The goal is demo cloud access, and do some simple calculation.

> suggestion would be to point to the [official pangeo docs for the data](https://pangeo-data.github.io/pangeo-cmip6-cloud/accessing_data.html#manually-searching-the-catalog),

Yeah this would be great! I think the xarray-tutorial should illustrate concepts and point out to any other resource we can.

**Comment by jbusecke** [comment]
That all sounds great!

**Comment by jbusecke** [comment]
> (cool repo name btw)

Obvi that was @TomNicholas s idea, cause I am not the best name-giver...

File: intermediate/remote_data_access.ipynb (hunk 1 above)
Comment (weiji14) [reply intermediate/remote_data_access.ipynb:1]: Good to know, I've updated the CSV link in e96181b89f369e1358ee421de0529cfee5429a48

Comment (weiji14) [reply intermediate/remote_data_access.ipynb:1]: Decided to go with `.squeeze()`, thanks for the tip @dcherian! Done in e96181b89f369e1358ee421de0529cfee5429a48.

**Comment by weiji14** [comment]
Thanks @jbusecke for the review (again)! Felt like it was too much of a coincidence that you've reviewed this PR and the PyOpenSci/PyGMT one on the same day, haha.

> > I would be happy to extend this example for more models/member
> 
> I think not necessary here; we could just link to your efforts at easy_ipcc (cool repo name btw). The goal is demo cloud access, and do some simple calculation.

Yes, looking at the ensemble is always nice, but probably a bit overkill for this tutorial. Thanks for the link to https://github.com/jbusecke/easy_ipcc though! Will be very helpful if I ever need to reproduce any of those plots!

> > suggestion would be to point to the [official pangeo docs for the data](https://pangeo-data.github.io/pangeo-cmip6-cloud/accessing_data.html#manually-searching-the-catalog),
> 
> Yeah this would be great! I think the xarray-tutorial should illustrate concepts and point out to any other resource we can.

Ok, I've pointed to the Pangeo/ESGF Cloud Data Working Group documentation link in 5711d9e6563661e54032f69b9771962947b172de. Hopefully that's ok.

Oh, and the notebook has also been added to the table of contents at 5711d9e6563661e54032f69b9771962947b172de. Let me know if any more changes are needed!

File: intermediate/remote_data_access.ipynb [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "eb263332-dc60-4bd1-9ef3-cf9612cf09a1",
+   "metadata": {},
+   "source": [
+    "Over 5 million rows! Let's filter it down to the variable and experiment\n",
+    "we're interested in, e.g. sea surface height.\n",
+    "\n",
+    "For the `variable_id`, you can look it up given some keyword at\n",
+    "https://docs.google.com/spreadsheets/d/1UUtoz6Ofyjlpx5LdqhKcwHFz2SGoTQV2_yekHyMfL9Y/edit#gid=1221485271\n",
```
Comment (weiji14) [inline intermediate/remote_data_access.ipynb:93]: Try removing the `edit#gid` part to see if it fixes CI.
```suggestion
    "https://docs.google.com/spreadsheets/d/1UUtoz6Ofyjlpx5LdqhKcwHFz2SGoTQV2_yekHyMfL9Y\n",
```

File: intermediate/cmip6-cloud.ipynb [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b3f5660d-bd46-44f6-8f6d-a62947b6f2c4",
+   "metadata": {},
+   "source": [
+    "In many cases, you'll need to first connect to the cloud provider.\n",
+    "The CMIP6 dataset allows anonymous access, but for some cases,\n",
+    "you may need to authentication."
```
Comment (dcherian) [inline intermediate/cmip6-cloud.ipynb:158]:
```suggestion
    "you may need to authenticate."
```

**Pull request merged**